conans/test/integration/command/runner_test.py

```diff
@@ -0,0 +1,222 @@
+import textwrap
+import os
+import pytest
+import docker
+from conans.test.utils.tools import TestClient
+from conans.test.assets.cmake import gen_cmakelists
+from conans.test.assets.sources import gen_function_h, gen_function_cpp
+
+
+
+def docker_skip(test_image=None):
+    try:
+        docker_client = docker.from_env()
+        if test_image:
+            docker_client.images.get(test_image)
+    except docker.errors.DockerException:
+        return True
+    except docker.errors.ImageNotFound:
+        return True
+    except docker.errors.APIError:
+        return True
+    return False
+
+
+def conan_base_path():
+    import conans
+    return os.path.dirname(os.path.dirname(conans.__file__))
+
+
+def dockerfile_path(name=None):
```

Comment:
One minor comment regarding the test style. I think it would be better to declare the file as a string here (or import the string from a common place once many tests in different places use it) and create the file using `client.save()`

Reply:
this is a static path inside the conan test folder.
For example: [conans/test/integration/command/dockerfiles/Dockerfile](https://github.com/conan-io/conan/pull/15856/files/2f1ca8e6ccc10998c7e497e3db7354c0268fb825#diff-c6cf0d0ee9a0571a63fea87b6ff5e4be8bafed126a327e79e48c8abe36535e7e)